lms/djangoapps/completion/api/v1/tests/test_views.py

```diff
@@ -1,139 +1,190 @@
+        ({
+            'course_key': COURSE_KEY,
             'blocks': {
-                'block-v1:edx+test+run+type@video+block@doggos': 1.0,
+                BLOCK_KEY: 1.0,
             }
-        }
-        request = self.factory.post(self.url, payload, format='json')
-        request.data = payload
-        force_authenticate(request, user=self.user)
-        response = self.view.post(request)
-        self.assertEqual(response.status_code, 400)
-
-    def test_batch_submit_with_no_blocks(self):
-        self.enroll()
-        payload = {
-            'username': self.user.username,
-            'course_key': self.course_key,
-        }
-        request = self.factory.post(self.url, payload, format='json')
-        request.data = payload
-        force_authenticate(request, user=self.user)
-        response = self.view.post(request)
-        self.assertEqual(response.status_code, 400)
+         }, 403, {"detail": 'You do not have permission to perform this action.'}),
+        # Course does not exist
+        ({
+            'username': ENROLLED_USERNAME,
+            'course_key': 'TestX/101/Test2',
+            'blocks': {
+                BLOCK_KEY: 1.0,
+            }
+         }, 404, {"detail": "CourseStructure matching query does not exist."}),
+    )
+    @ddt.unpack
+    def test_batch_submit(self, payload, expected_status, expected_data):
+        """
+        Test the batch submission response for student users.
+        """
+        response = self.client.post(self.url, payload, format='json')
+        self.assertEqual(response.data, expected_data)
+        self.assertEqual(response.status_code, expected_status)
+
+    @ddt.data(
+        # Staff can submit completion on behalf of other users
+        ({
+            'username': ENROLLED_USERNAME,
+            'course_key': COURSE_KEY,
+            'blocks': {
+                BLOCK_KEY: 1.0,
+            }
+         }, 200, {'detail': 'ok'}),
+        # User must be enrolled in the course
+        ({
+            'username': UNENROLLED_USERNAME,
+            'course_key': COURSE_KEY,
+            'blocks': {
+                BLOCK_KEY: 1.0,
+            }
+         }, 400, {"detail": "User is not enrolled in course."}),
+        # Username is required
+        ({
+            'course_key': COURSE_KEY,
+            'blocks': {
+                BLOCK_KEY: 1.0,
+            }
+         }, 400, {"detail": "Key 'username' not found."}),
+        # User must not exist
+        ({
+            'username': 'doesntexist',
+            'course_key': COURSE_KEY,
+            'blocks': {
+                BLOCK_KEY: 1.0,
+            }
+         }, 404, {"detail": 'User matching query does not exist.'}),
+    )
+    @ddt.unpack
+    def test_batch_submit_staff(self, payload, expected_status, expected_data):
+        """
+        Test the batch submission response when logged in as a staff user.
+        """
+        self.client.force_authenticate(user=self.staff_user)
```

Comment:
Also added tests to ensure the staff user can submit completions for any learner, and to ensure coverage of all the error cases.